### DISCUSSION TITLE
Update content outside of swup container

### DISCUSSION THREAD

**Discussion by ShaneSheppard** [swup/swup#818] [Help] [answered]
Hello Again.

I'm using swup, Parallel Plugin & Head Plugin.

I have next and previous links outside of my swup container so they don't show during the transition between pages and look the same on every page.

So the next and previous buttons and the fixed navigation.

The next and previous buttons are page urls with the "data-swup-animation" custom attribute left or right

The links i set on the next and previous buttons are being rewritten with the current page url followed by a # after the first initial page load.

What do i need to do to stop urls from being rewritten when linking out side of the container?

```
<div class="navbar"></div>
<div class="page_buttons">
<!-- Links to previous and next pages manually set -->
</div>
<div class="page"> <!-- transition css grid etc -->
<div class="transition-slide" id="swup"><!-- container for swup -->
<!-- Page Content -->
</div>
</div>
```

**Comment by daun** [comment]
What do you mean by "urls being rewritten" — the browser's location bar updating to the next page's url? And how are you updating the prev/next buttons if they're outside of the main container? And why the `#` after the url after the first page load? So many questions 🤠

**Reply by ShaneSheppard** [reply]
@daun Here is my test page - 
...

**Reply by ShaneSheppard** [reply]
it transitions perfectly when going to the next page (blue page) but when going back (red page)  it's rewritten my link with the current page url proceeded by a hash so i can't navigate back to the red page.

**Reply by daun** [reply]
The previous page link is missing a url — it's href is hardcoded to `#`, so that's what swup will try and load. If you want to navigate back to the red page, you'll have to update the link to point to that url. swup doesn't know what the previous/next page is.

**Reply by ShaneSheppard** [reply]
That's my point it's not hard coded, in my markup it's set to /tests/red-test on the previous button.

**Reply by daun** [reply, marked as answer]
You'll need to populate the prev/next links with the correct URLs in webflow, and then update the links after each swup navigation so they keep pointing at the correct urls. You could look into the [Morph Plugin](https://github.com/daun/swup-morph-plugin) to update the links. In any case, the links' `href` attributes must be set in your html markup.

**Reply by daun** [reply]
This is the markup I'm seeing for those buttons:

```html
<a data-swup-animation="left" href="#" class="page_button_left w-inline-block"></a>
<a data-swup-animation="right" href="/tests/blue-test" class="page_button_right w-inline-block">
```

**Reply by ShaneSheppard** [reply]
If you load this page - 
...
You can see it's not hard coded as a # 
It's only after the first page transitions that the url is then hashed with the current page.

**Reply by ShaneSheppard** [reply]
I will try the Morph plugin and come back to you.

**Reply by daun** [reply]
The markup I posted above is what's coming back as html from the server, before any page transitions are run. For the other url you posted, it's just the other way around, with the `#` in the next page's button. You can check the Network tab of your browser's developer tools to see what's coming back.

**Comment by ShaneSheppard** [comment]
> You'll need to populate the prev/next links with the correct URLs in webflow, and then update the links after each swup navigation so they keep pointing at the correct urls. You could look into the [Morph Plugin](https://github.com/daun/swup-morph-plugin) to update the links. In any case, the links' `href` attributes must be set in your html markup.

Yes morph was the answer around the buttons and nav conatiners
